### DISCUSSION TITLE
Oh... This is Prod?! - Gunnar Morling

### DISCUSSION THREAD

**Discussion by giscus** [gunnarmorling/discussions.morling.dev#305] [Announcements]
# Oh... This is Prod?! - Gunnar Morling

I strongly believe that you should avoid connecting to production environments from local developer machines as much as possible.
But sometimes, e.g. in order to analyse some specific kinds of failures,
doing so can be inevitable.

Now, if this is the case, I really, really want to be sure that I’m aware of the environment I am working in.
I absolutely want to avoid a situation as in the catchy title of this post,
when for instance you realize that you just ran some integration test against a production environment.
In the context of working with the AWS CLI tool this means I’d like to be aware of the currently active profile by means of coloring my shell accordingly.
Here’s how I’ve set this up using iTerm2 and zsh.

https://www.morling.dev/blog/oh_this_is_prod/

**Comment by Grandys** [comment]
Similar "mistake" can happen with database. Let's say one wants to clean up data before running integration tests, runs `TRUNCATE TABLE ..` in PGadmin, "hmm, why it takes so long. Oops, that console wasn't connected to local database...".  
Personally, I have a rule to never save credentials to non-local database in clients (Datagrid/PgAdmin). Every attempt to execute statement against non local environment will prompt for password.
